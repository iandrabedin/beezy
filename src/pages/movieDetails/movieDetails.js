import React, { useEffect, useState, lazy, Suspense } from "react";
import { useParams } from "react-router-dom";
import { fetchDetails, fetchCredits } from "../../api";
import { Loading } from "../components";

const Details = lazy(() => import("../components/details"));

export const MovieDetails = () => {
  let { movieId } = useParams();
  // States
  const [movieDetails, setMovieDetails] = useState([]);
  const [credits, setCredits] = useState([]);

  // Effects
  useEffect(() => {
    fetchDetails(movieId).then((result) => setMovieDetails(result));
    fetchCredits(movieId).then((result) => setCredits(result));
  }, [movieId]);

  return (
    <Suspense fallback={<Loading />}>
      <Details movieDetails={movieDetails} credits={credits} />
    </Suspense>
  );
};
