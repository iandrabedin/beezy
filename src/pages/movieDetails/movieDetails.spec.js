import React from "react";
import { cleanup, render } from "@testing-library/react";
import { MovieDetails } from "./movieDetails";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    movieId: 270
  })
}));

describe("MovieDetails", () => {
  it("should render as expected", () => {
    const { asFragment } = render(<MovieDetails />);
    expect(asFragment()).toMatchSnapshot();
  });
});
