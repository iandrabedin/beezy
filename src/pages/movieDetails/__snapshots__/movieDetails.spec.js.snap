// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render as expected 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    Loading...
  </div>
</DocumentFragment>
`;
